File: construct-string-from-binary-tree/construct-string-from-binary-tree.py


class Solution:
    def tree2str(self, root: Optional[TreeNode]) -> str:
        if not root: return ""
        ans = str(root.val)
        if root.left: ans += "(" + self.tree2str(root.left) + ")"
        elif root.right: ans += "()"
        if root.right: ans += "(" + self.tree2str(root.right) + ")"
        return ans

        # Construct String

File: construct-string-from-binary-tree/construct-string-from-binary-tree.java

 */
class Solution {
    public String tree2str(TreeNode root) {
        if(root == null) return "";
        StringBuilder ans = new StringBuilder();
        ans.append(root.val);
        if(root.left!=null) ans.append("(").append(tree2str(root.left)).append(")");
        else if(root.right != null) ans.append("()");
        if(root.right!=null) ans.append("(").append(tree2str(root.right)).append(")");
        return ans.toString();
    }
}

// Construct String